### ISSUE TITLE
[PubSub] PubSubInboundChannelAdapter stop pulling messages from PubSub

### ISSUE THREAD

**Issue by brachipa** [spring-cloud/spring-cloud-gcp#2552]
I have SpringBoot App deployed in GKE pod, that pull messages from pubsub: 
```java
 PubSubInboundChannelAdapter adapter = new PubSubInboundChannelAdapter(pubSubTemplate, subscription);
 logger.info("Created subscription to " + subscription);	
 adapter.setOutputChannel(inputChannel);	
 adapter.setAckMode(AckMode.MANUAL);
```
It works fine, but then it stopped pulled messages, if I restart the pod it works fine.
Nothing in the logs.
I have many pods that subscribed to different subscriptions and this issue happens for all of them.
can be once in a month, but consistently happens, and require us to restart the pods all the time (production cluster)

spring boot: 2.3.0.RELEASE
spring-cloud-gcp-starter-pubsub  1.2.5.RELEASE

**Comment by elefeint** [comment]
@brachipa When the listener stops, is there an unusually long amount of time that passed from when the last message was sent to the topic?

Do you override `spring.cloud.gcp.pubsub.keepAliveIntervalMinutes` property to any custom value?

Do you override any of the Spring Cloud GCP pub/sub subscription beans -- `subscriberTransportChannelProvider`, `SubscriberFactory`, `PubSubSubscriberTemplate`?

**Comment by meltsufin** [comment]
@brachipa Do you have other pods that are getting the messages at the same time? Perhaps this node is just being starved?
Since Spring  Cloud GCP relies on the Pub/Sub client library, it's possible that there is an issue there.
cc/ @chingor13

**Comment by brachipa** [comment]
Yes, I also forget to mention that I have 2 subscriptions in the same pod, one is getting message in the same time and one is totally disconnected.

**Comment by meltsufin** [comment]
I would try the [Synchronous Pull](https://docs.spring.io/spring-cloud-gcp/docs/1.2.5.RELEASE/reference/html/#pollable-message-source-using-pubsub-synchronous-pull) method, which is less prone to starvation, and you have more control over.
